redash/tasks/failure_report.py

```diff
@@ -0,0 +1,65 @@
+import datetime
+from collections import Counter
+from flask import render_template
+from redash.tasks.general import send_mail
+from redash.worker import celery
+from redash import redis_connection, settings
+from redash.utils import json_dumps, json_loads, base_url
+
+
+@celery.task(name="redash.tasks.send_aggregated_errors")
+def send_aggregated_errors(email_address):
+    key = 'aggregated_failures:{}'.format(email_address)
+    errors = [json_loads(e) for e in redis_connection.lrange(key, 0, -1)]
+
+    if errors:
+        errors.reverse()
+        occurrences = Counter((e.get('id'), e.get('message')) for e in errors)
+        unique_errors = {(e.get('id'), e.get('message')): e for e in errors}
+
+        context = {
+            'failures': [{
+                    'base_url': v.get('base_url'),
+                    'id': v.get('id'),
+                    'name': v.get('name'),
+                    'failed_at': v.get('failed_at'),
+                    'failure_reason': v.get('message'),
+                    'failure_count': occurrences[k],
+                    'comment': v.get('comment')
+            } for k, v in unique_errors.iteritems()]
+        }
+
+        html = render_template('emails/failures.html', **context)
+        text = render_template('emails/failures.txt', **context)
+        subject = "Redash failed to execute {} of your queries".format(len(unique_errors.keys()))
+        send_mail.delay([email_address], subject, html, text)
+
+    redis_connection.delete(key)
+
+
+def notify_of_failure(message, query):
+    if not settings.organization.SEND_EMAIL_ON_FAILED_SCHEDULED_QUERIES:
+        return
```

Comment:
Another check to perform is if the query owner is disabled. If they are, then no need to send anything.

Reply:
Do disabled users' scheduled queries continue to run? Shouldn't they be disabled or at least [skipped](https://github.com/getredash/redash/commit/b5b56430904a17819916d64f5d3b950d0a82d756#diff-5ca0dc869c9c3218d8a95ba7f99f5f4cR278)?

Reply:
Many times although the user left the company (or was disabled for whatever reason) their content (queries, dashboards) are still used.

Reply:
👍  7b232be